### ISSUE TITLE
Questions on pressure iteration for fine-coarse meshes.

### ISSUE THREAD

**Issue by marcosvanella** [firemodels/fds#10966]
Kevin, as per email exchange:

While reviewing the PRESSURE_ITERATION_SCHEME I noted that in the CHECK_WALL_LOOP in COMPUTE_VELOCITY_ERROR (pres.f90) the forcing in WALL_WORK1 is not defined for "small" wall cells of a fine-coarse mesh interface:

CHECK_WALL_LOOP: DO IW=1,N_EXTERNAL_WALL_CELLS+N_INTERNAL_WALL_CELLS

   WC=>WALL(IW)
   ...

   IF (WC%BOUNDARY_TYPE==INTERPOLATED_BOUNDARY) THEN
      EWC=>EXTERNAL_WALL(IW)
      IF (EWC%AREA_RATIO<0.9_EB) CYCLE CHECK_WALL_LOOP
      OM => OMESH(EWC%NOM)
      M2 => MESHES(EWC%NOM)
   ENDIF
...

A question that comes to mind is what's the reason for this? Seems we are letting the coarse side WALL_WORK1 drive the value of H to be used in the fine-coarse boundary BC.
I also noted that later a similar test is done in
...
   ! If the grid cells in the current mesh are smaller than those of the other mesh, do not include in error tolerance

   IF (WC%BOUNDARY_TYPE==INTERPOLATED_BOUNDARY) THEN
      IF (OM%NIC_R>OM%NIC_S) CYCLE CHECK_WALL_LOOP
   ENDIF
...

to avoid adding the fine side errors in the VEL_ERROR_MAX calculation. This seems to me redundant as these wall cells have been skipped before?

**Comment by mcgratta** [comment]
In answer to the first question -- we cannot check the convergence of the normal component of velocity for the smaller grid cell at the mesh interface. There is no comparable velocity component in the other coarse mesh to compare it to. Second question -- why this extra IF block? There is no reason for it. I removed it. Now, we expect the normal velocity on the coarse mesh to converge to the average of the normal velocities on the fine. I tested this with the dancing_eddies cases and it seems OK to check for convergence even at the mesh interface of two different grid resolutions.
